server/src/db/post_view.rs

```diff
@@ -73,81 +75,35 @@ pub struct PostView {
   pub saved: Option<bool>,
 }
 
-impl PostView {
-  pub fn list(
-    conn: &PgConnection,
-    type_: ListingType,
-    sort: &SortType,
-    for_community_id: Option<i32>,
-    for_creator_id: Option<i32>,
-    search_term: Option<String>,
-    url_search: Option<String>,
-    my_user_id: Option<i32>,
+pub struct PostViewQuery<'a> {
+  conn: &'a PgConnection,
+  query: BoxedQuery<'a, Pg>,
+  my_user_id: Option<i32>,
+  for_creator_id: Option<i32>,
+  page: Option<i64>,
+  limit: Option<i64>,
+}
+
+impl<'a> PostViewQuery<'a> {
+  pub fn create(
+    conn: &'a PgConnection,
+    r#type: ListingType,
```

Comment:
I've never seen vars named like this, what does this do?

Reply:
It allows you to name the field `type`, as it is a reserved keywork